examples/benchmark.py

```diff
@@ -31,183 +124,469 @@ def downloadAmberSuite():
+    else:
+        friction = 91*(1/unit.picoseconds)
+    if amoeba:
+        dt = 0.002*unit.picoseconds
+        if options.ensemble == 'NVE':
+            integ = mm.MTSIntegrator(dt, [(0,2), (1,1)])
+        else:
+            integ = mm.MTSLangevinIntegrator(temperature, friction, dt, [(0,2), (1,1)])
+    elif amber:
+        dt = 0.004*unit.picoseconds
+        if options.ensemble == 'NVE':
+            integ = mm.VerletIntegrator(dt)
+        else:
+            integ = mm.LangevinMiddleIntegrator(temperature, friction, dt)
+    else:
+        if options.bond_constraints == 'hbonds':
             dt = 0.004*unit.picoseconds
-            constraints = app.HBonds
-            hydrogenMass = 1.5*unit.amu
             if options.ensemble == 'NVE':
                 integ = mm.VerletIntegrator(dt)
             else:
                 integ = mm.LangevinMiddleIntegrator(temperature, friction, dt)
-        positions = pdb.positions
-        system = ff.createSystem(pdb.topology, nonbondedMethod=method, nonbondedCutoff=cutoff, constraints=constraints, hydrogenMass=hydrogenMass)
-    if options.ensemble == 'NPT':
-        system.addForce(mm.MonteCarloBarostat(1*unit.bar, temperature, 100))
-    print('Step Size: %g fs' % dt.value_in_unit(unit.femtoseconds))
+        elif options.bond_constraints == 'allbonds':
+            dt = 0.005*unit.picoseconds
+            hydrogenMass = 4*unit.amu
+            if options.ensemble == 'NVE':
+                integ = mm.VerletIntegrator(dt)
+            else:
+                integ = mm.LangevinIntegrator(temperature, friction, dt)
+        else:
+            raise ValueError(f'Unknown bond_constraints {options.bond_constraints}')
+
+    test_result['timestep_in_fs'] = dt.value_in_unit(unit.femtoseconds)
     properties = {}
     initialSteps = 5
-    if options.device is not None and platform.getName() in ('CUDA', 'OpenCL'):
+    if options.device is not None and options.platform in ('CUDA', 'OpenCL'):
         properties['DeviceIndex'] = options.device
         if ',' in options.device or ' ' in options.device:
             initialSteps = 250
-    if options.precision is not None and platform.getName() in ('CUDA', 'OpenCL'):
+    if options.precision is not None and options.platform in ('CUDA', 'OpenCL'):
         properties['Precision'] = options.precision
 
-    # Run the simulation.
-    
+    # Add barostat if requested
+    if options.ensemble == 'NPT':
+        system.addForce(mm.MonteCarloBarostat(1*unit.bar, temperature, 100))
+
+    # Create the Context
+    platform = mm.Platform.getPlatformByName(options.platform)
     integ.setConstraintTolerance(1e-5)
     if len(properties) > 0:
         context = mm.Context(system, integ, platform, properties)
     else:
         context = mm.Context(system, integ, platform)
+
+    # Store information about the Platform used by the Context
+    platform = context.getPlatform()
+    test_result['precision'] = options.precision # requested precision
```

Comment:
Only store precision for CUDA and OpenCL.

Reply:
Even better, we can check `'Precision' in platform.getPropertyNames()` to see whether the selected platform supports precision.  That way we don't need to hardcode a list of platforms.

Reply:
You forgot to remove this line when you added the check below.